### DISCUSSION TITLE
Data entry problem

### DISCUSSION THREAD

**Discussion by SDGArmenia** [open-sdg/open-sdg#2001] [Q&A] [answered]
After updating the SDG version, there was a problem with data enyty. Unable to enter data through the interface.
Could somebody help  to solve the problem.

https://armstat.github.io/sdg-site-armenia/2-2-2/data/

Best Regards.

**Comment by Govej** [comment]
@SDGArmenia This was a bug in previous versions of Open SDG, in order to fix this we would advise upgrading your site to the latest version (2.2.0)

Instructions on how to upgrade and details on new features can be found here

https://open-sdg.readthedocs.io/en/latest/upgrades/upgrading-2-2-0/

**Comment by SDGArmenia** [comment]
@Govej Thank you for your prompt response. 
We upgraded data repository to sdg-build 2.2.0,  but when we try to update site repository's _config.yml file 
https://github.com/armstat/sdg-site-armenia/blob/develop/_config.yml

          remote_theme: open-sdg/open-sdg@2.2.0   

we get the following failed deployment in Circleci.



--**Test the HTML**---------------------------------------------------------------
#!/bin/bash -eo pipefail
bash scripts/test/html_proofer_staging.sh
NOTE: nokogumbo: Using Nokogiri::HTML5 provided by Nokogiri. See https://github.com/sparklemotion/nokogiri/issues/2205 for more information.
Running ["ScriptCheck", "LinkCheck", "ImageCheck"] on ["./_test"] on *.html... 

Ran on 4000 files!

- ./_test/sdg-site-armenia/1-4-2/index.html
  *  internally linking to /sdg-site-armenia/NA, which does not exist (line 758)
     <a href="/sdg-site-armenia/NA">
                United Nations Sustainable Development Goals Metadata (PDF 4.0 MB)
              </a>
- ./_test/sdg-site-armenia/am/1-4-2/index.html
  *  internally linking to /sdg-site-armenia/NA, which does not exist (line 758)
     <a href="/sdg-site-armenia/NA">
                United Nations Sustainable Development Goals Metadata (PDF 4.0 MB)
              </a>
- ./_test/sdg-site-armenia/ru/1-4-2/index.html
  *  internally linking to /sdg-site-armenia/NA, which does not exist (line 758)
     <a href="/sdg-site-armenia/NA">
                United Nations Sustainable Development Goals Metadata (PDF 4.0 MB)
              </a>
HTML-Proofer found 3 failures!
Exited with code exit status 1
CircleCI received exit code 1
------------------------------------------------------------------------------------------



In Gemfile  we have

source "https://rubygems.org"

gem "jekyll", "3.8.4"
gem "html-proofer"
gem "jekyll-remote-theme"
gem "deep_merge"
gem "jekyll-open-sdg-plugins", "2.2.0"

in config.yml file of the .circleci folder we have

    test:
    docker:
      - image: circleci/ruby:2.5.1
    working_directory: ~/repo


and 

 docker:
      - image: circleci/ruby:2.5.1
    working_directory: ~/repo


Must we upgrade the version 3.8.4 of jekyll  or version 2.5.1 of  ruby: for sdg 2.2.0  and to which version must we upgrade?


How can we  solve this problem?

Best Regards.

**Reply by brockfanning** [reply]
@SDGArmenia In this case, the "html-proofer" test appears to be pointing out a broken link on indicator 1.4.2. So it's actually being helpful here. :) You should be able to fix the root of the problem, which I think is in the [indicator configuration/metadata for 1.4.2](https://github.com/armstat/sdg-data-armenia/blob/develop/meta/1-4-2.md). Right now that file includes:

```
goal_meta_link: NA
goal_meta_link_text: United Nations Sustainable Development Goals Metadata (PDF 4.0 MB)
```

If you are would prefer not to display the "Link to UN metadata" field, you could simply remove both of those lines.

Or, if you would prefer to display a link, you could change `NA` to an actual URL. For example:

```
goal_meta_link: https://unstats.un.org/sdgs/metadata/files/Metadata-01-04-02.pdf
goal_meta_link_text: United Nations Sustainable Development Goals Metadata (PDF 4.0 MB)
```

## Important note

Separately, I would also recommend changing [the Gemfile line for "html-proofer"](https://github.com/armstat/sdg-site-armenia/blob/develop/Gemfile#L4). Try changing that line to:

```
gem "html-proofer", "3.19.4"
```

We have found that the more recent versions of "html-proofer" do not work well with Open SDG, and so we recommend using a specific version of html-proofer (3.19.4).

**Reply by SDGArmenia** [reply]
@brockfanning  Thank you for help .We have added a link and solved the problem, but when we have changed   "html-proofer" to the " html-proofer (3.19.4)".
source "https://rubygems.org"

gem "jekyll", "3.8.4"
gem "html-proofer", "3.19.4"
gem "jekyll-remote-theme"
gem "deep_merge"
gem "jekyll-open-sdg-plugins", "2.2.0"

we heve got another problem with install Ruby dependencies

-----install Ruby dependencies-----
#!/bin/bash -eo pipefail
bundle check --path=vendor/bundle || bundle install --path=vendor/bundle
Bundler can't satisfy your Gemfile's dependencies.
Install missing gems with `bundle install`.
Fetching gem metadata from https://rubygems.org/..........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby

    html-proofer (= 3.19.4) was resolved to 3.19.4, which depends on
      ruby (< 4.0, >= 2.6.0)


-------------------------------

We have jekyll version 3.8.4 and ruby version 2.5.1: which versions should we upgrade to for sdg 2.2.0?

Best Regards.

**Reply by brockfanning** [reply, marked as answer]
@SDGArmenia The environment needed to be upgraded to Ruby 2.6. Now that that has been done, can you please try the html-proofer change again?

**Reply by SDGArmenia** [reply]
Thank you very much. We changed the html-proofer and now everything seems to be fine.
